logdna/client_test.go

```diff
@@ -0,0 +1,57 @@
+package logdna
+
+import (
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var url = "https://api.logdna.com"
+
+func TestClient_View(t *testing.T) {
+	payload := ViewPayload{Name: "test", Query: "test"}
+	client := Client{ServiceKey: servicekey, HTTPClient: &http.Client{Timeout: 15 * time.Second}}
+	result, err := client.CreateView(url, payload)
+	assert.Equal(t, nil, err)
+	assert.Len(t, result, 10)
+}
+
+func TestClient_Alert(t *testing.T) {
+	var channels []Channel
+	channels = append(channels, Channel{Integration: "pagerduty", Key: "Your PagerDuty API key goes here", TriggerInterval: "15m", TriggerLimit: 20})
+	payload := ViewPayload{Name: "test", Channels: channels}
+	client := Client{ServiceKey: servicekey, HTTPClient: &http.Client{Timeout: 15 * time.Second}}
+	result, err := client.CreateAlert(url, payload)
+	assert.Equal(t, nil, err)
+	assert.Len(t, result, 10)
+}
+
+func TestClient_ViewURLError(t *testing.T) {
+	payload := ViewPayload{Name: "test", Query: "test"}
+	client := Client{ServiceKey: servicekey, HTTPClient: &http.Client{Timeout: 15 * time.Second}}
+	_, err := client.CreateView("https://api.logdna.co", payload)
+	assert.Equal(t, errors.New("Error with view: Post \"https://api.logdna.co/v1/config/view\": dial tcp: lookup api.logdna.co on 127.0.0.11:53: no such host"), err)
+}
+
+func TestClient_AlertURLError(t *testing.T) {
+	payload := ViewPayload{Name: "test"}
+	client := Client{ServiceKey: servicekey, HTTPClient: &http.Client{Timeout: 15 * time.Second}}
+	_, err := client.CreateAlert("https://api.logdna.co", payload)
+	assert.Equal(t, errors.New("Error with alert: Post \"https://api.logdna.co/v1/config/alert\": dial tcp: lookup api.logdna.co on 127.0.0.11:53: no such host"), err)
+}
+func TestClient_ViewResourceNotFound(t *testing.T) {
+	payload := ViewPayload{Name: "test", Query: "test"}
+	client := Client{ServiceKey: servicekey, HTTPClient: &http.Client{Timeout: 15 * time.Second}}
+	_, err := client.CreateView("https://api.logdna.com/v1/config/invalidresource", payload)
```

Comment:
again here, instead of hitting the real API the test server you create here would return a "invalid resource" response. you can think of this pattern with the `httptest` package as being similar to how we would use `nock` with node.